prov/util/src/util_wait.c

```diff
@@ -292,12 +292,8 @@ static int util_wait_fd_control(struct fid *fid, int command, void *arg)
 	wait = container_of(fid, struct util_wait_fd, util_wait.wait_fid.fid);
 	switch (command) {
 	case FI_GETWAIT:
-#ifdef HAVE_EPOLL
 		*(int *) arg = wait->epoll_fd;
 		ret = 0;
-#else
-		ret = -FI_ENOSYS;
-#endif
```

Comment:
This path is called to obtain a single fd that can be used by the caller to wait for events.  The fi_epoll_xxx functions are internal to libfabric, so only internal users can make use of it.  Can you add more details in the commit message what this change is for?  It does not work in a generic sense, where an external application is making this call.

Reply:
Yes, this is exactly to wait for events (for example, fi_cntr_wait) in external application. since util (and rxm) supports only epoll, and Windows does not support epoll, we need to completely enable epoll->poll fallback in util provider. This is the only place where HAVE_EPOLL still exists and restricts functionality on systems w/o epoll. I've updated commit message.

Reply:
An external application can't make use of this without using the same implementation that libfabric uses internally.  That implementation isn't guaranteed not to change between versions.

When epoll is available, OFI uses an fi_epoll_t that maps to an int, which is a real fd.  When epoll is not available, fi_epoll_t is a pointer to a structure, which we can't even return to the user, since we're trying to pass it as an int.